src/lexer.coffee

```diff
@@ -1199,10 +1199,12 @@ CSX_ATTRIBUTE = /// ^
 ///
 
 NUMBER     = ///
-  ^ 0b[01]+    |              # binary
-  ^ 0o[0-7]+   |              # octal
-  ^ 0x[\da-f]+ |              # hex
-  ^ \d*\.?\d+ (?:e[+-]?\d+)?  # decimal
+  ^ 0b[01](?:_?[01])*        |    # binary
+  ^ 0o[0-7](?:_?[0-7])*      |    # octal
+  ^ 0x[\da-f](?:_?[\da-f])*  |    # hex
+  # ^ \d*\.?\d+ (?:e[+-]?\d+)?    # decimal
```

Comment:
Why is this commented out?

Reply:
@GeoffreyBooth The next 2 lines cover that pattern. I kept it there as a simplified / more readable version for future reference.

Reply:
Can you please explain this in a comment? This just looks like commented-out code, and it’s unclear what the next two lines are.

Reply:
Done.